sphinx/domains/python.py

```diff
@@ -285,12 +290,54 @@ def add_target_and_index(self, name_cls, sig, signode):
                                               fullname, '', None))
 
     def before_content(self):
-        # needed for automatic qualification of members (reset in subclasses)
-        self.clsname_set = False
+        # type: () -> None
+        """Handle object nesting before content
+
+        If this class is a nestable object, such as a class object, build up a
```

Comment:
I felt confused the description uses "class" twice. The first class; "this class" means subclasses of `PyObject`. It is an implementation of directives. so my proposal was wrong.
And second one; "a class object" means subjects of directives. Sometime it is a class, a method, and so on.

I'm not a native English speaker and not good at English. so If you feel this is okay, go ahead with this.